sys/arm64/arm64/elf_machdep.c

```diff
@@ -752,7 +752,7 @@ elf_reloc_self(const Elf_Dyn *dynp, void *data_cap, const void *code_cap)
 	for (; dynp->d_tag != DT_NULL; dynp++) {
 		switch (dynp->d_tag) {
 		case DT_RELA:
-			rela = (const Elf_Rela *)((const char *)data_cap +
+			rela = (const Elf_Rela *)cheri_setaddress(data_cap,
```

Comment:
Hm, this is wrong for a PIE kernel, but I guess we don't have a relocbase equivalent here to add in for that?

Reply:
I'm assuming we only have a PDE kernel.  I've considered trying to support PIE kernels but that seems like a fair bit more work.  I think the way I'd better like to support PDE kernels even is more what we do in kldxref where we pass in a pointer to the start of the file and a base VM address that is more like a relocbase in reverse, so instead of `relocbase + <file VA>` you do `mapbase + (<file VA> - base_va)`, but that's a more invasive change in terms of API churn affecting elf_reloc, etc.